mitiq/zne/scaling/identity_insertion.py

```diff
@@ -0,0 +1,425 @@
+        )
+
+
+def _create_scale_mask(
+    weight_mask: List[float],
+    scale_factor: float,
+    scaling_method: str,
+    seed: Optional[int] = None,
+) -> List[float]:
+    r"""Returns a list of integers determining how many times an identity gate
+    must be inserted to realize the desired input scale_factor.
+    More precicely, the j_th element of the returned list is associated
+    to the j_th gate G_j of a circuit that we want to scale and determines
+    how many times I should be applied after G_j.
+    The gate ordering is equal to the one used in the `all_operations()`
+    method of the :class:`cirq.Circuit` class: gates from earlier moments
+    come first and gates within the same moment follow the order in which
+    they were given to the moment's constructor.
+    The returned list is built such that the total weight of the
+    folded circuit is approximately equal to scale_factor times the
+    total weight of the input circuit.
+    For equal weights, this function reproduces identity scaled version of
+    the local unitary folding method defined in equation (5) of
+    :cite:`Giurgica_Tiron_2020_arXiv`.
+    Args:
+        weight_mask: The weights of all the gates of the circuit to fold.
+            Highly noisy gates should have a corresponding high weight.
+            Gates with zero weight are assumed to be ideal and are not folded.
+        scale_factor: The effective noise scale factor.
+        scaling_method: A string equal to "at_random", or "from_left", or
+            "from_right". Determines the partial folding method described in
+            :cite:`Giurgica_Tiron_2020_arXiv`. If scale_factor is an odd
+            integer, all methods are equivalent and this option is irrelevant.
+        seed: A seed for the random number generator. This is used only when
+            scaling_method is "at_random".
+    Returns: The list of integers determining to how many times one should
+            insert I in the circuit to be scaled.
+    Example:
+        >>>_create_scale_mask(
+            weight_mask=[1.0, 0.5, 2.0, 0.0],
+            scale_factor=4,
+            scaling_method="from_left",
+        )
+        [2, 2, 1, 0]
+    """
+
+    if not 1.0 <= scale_factor:
+        raise ValueError(
+            f"Requires scale_factor >= 1.0 but scale_factor = {scale_factor}."
+        )
+
+    # Find the maximum odd integer smaller or equal to scale_factor
+    num_uniform_folds = int(scale_factor - 1.0)
+    odd_integer_scale_factor = np.abs(num_uniform_folds - 1)
+
+    # Uniformly scaling all gates to reach odd_integer_scale_factor
+    num_inserts_mask = []
+    for w in weight_mask:
+        if np.isclose(w, 0.0):
+            num_inserts_mask.append(0)
+        else:
+            num_inserts_mask.append(num_uniform_folds)
+
+    # If the scale_factor is an odd integer, we are done.
+    if np.isclose(odd_integer_scale_factor, scale_factor, atol=0.01):
+        return num_inserts_mask
```

Comment:
I think this  should be corrected in this way. More or less, I didn't test.

```suggestion
    # Find the maximum integer smaller or equal to scale_factor
    num_uniform_inserts = int(scale_factor - 1.0)
    integer_scale_factor = num_uniform_inserts + 1

    # Uniformly scaling all gates to reach odd_integer_scale_factor
    num_inserts_mask = []
    for w in weight_mask:
        if np.isclose(w, 0.0):
            num_inserts_mask.append(0)
        else:
            num_inserts_mask.append(num_uniform_inserts)

    # If the scale_factor is an integer, we are done.
    if np.isclose(integer_scale_factor, scale_factor, atol=0.01):
        return num_inserts_mask
```

Reply:
@andreamari Yes, I tried using `num_uniform_inserts + 1` but there was an incompatibility with `scale_factor`. 
For identity scaling, `scale_factor=1` would insert 1 identity gate whereas `scale_factor=0` would keep the circuit unchanged. This was different from how `scale_factor=1` is used in unitary folding. 

Can I use `insert_factor` instead of `scale_factor` ? See above discussion.

Reply:
> @andreamari Yes, I tried using `num_uniform_inserts + 1` but there was an incompatibility with `scale_factor`.
> For identity scaling, `scale_factor=1` would insert 1 identity gate whereas `scale_factor=0` would keep the circuit unchanged. This was different from how `scale_factor=1` is used in unitary folding.

Are you sure?   If `scale_factor = 1`, `num_uniform_inserts` is 0 and `integer_scale_factor` is 1.  So `num_inserts_mask` should be `[0, 0, .... 0]`.  Am I wrong? 


> Can I use `insert_factor` instead of `scale_factor` ? See above discussion.
The optimal solution would be to use `scale_factor` for consistency with other folding functions. But if this is really necessary we can discuss about it.